### ISSUE TITLE
If trying to plant dynamite on friendly objective you should get filled chargebar

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#597]
Author: @saukko28 (Saukko)
Date: 2014-10-10
Redmine Issue: 597, https://dev.etlegacy.com/issues/597

---

I got an example here. When you try to plant landmines on places where you can't it will fill your chargebar when you try to arm them. Also the landmine will disappear during the progress. But in dynamite's case, there's not this feature. Instead of giving back your charge into chargebar it let's it be at the state when you threw the dynamite. So my suggestion is clear, when you try to plant dynamite at your own objective, your dynamite will disappear (as it does now) + you will get your chargebar filled again.

-*S

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2014-10-12T09:29:06Z

---

It's been discussed before on IRC, but imho dynamite and landmines are two different cases: landmine can't be armed everywhere due to world's limitations, while dynamite can't be armed near a friendly objective (but an enemy could still arm your dynamite).

Thus, I don't see any reason why the charge bar should be replenished when a dynamite is used, armed or not.

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2014-10-12T10:48:49Z

---

Spyhawk wrote:
> Thus, I don't see any reason why the charge bar should be replenished when a dynamite is used, armed or not.
I'm thinking the same.

Spyhawk (IRC) wrote:
>If an engi wants to blow up his own objectives, he should be punished.
I don't know if it's possible in all mods but some engineers want to protect the objective by dynamite. I hate this - several times I did protect the objective I got killed by team dynamite. Warbell construction on top of the roof is best example so yes - he/she should be punished :)

I don't think we should spend gimmicks for stupidness - each experienced engineer knows planting/arming defuses the chargebar. Let's close this issue.
